pkgs/build-support/make-darwin-app/default.nix

```diff
@@ -0,0 +1,74 @@
+{
+  lib,
+  stdenv,
+  makeWrapper,
+  undmg,
+}:
+
+# Returns the makeDarwinApp function
+{
+  /** ${appName}.app */
+  appName,
+  /** Name of the file found in ${appName}.app/Contents/MacOS/${binaryName}
+    * to be executed
+    * TODO: detect this automatically from the .plist file?
+    */
+  binaryName,
+  pname,
+  version,
+  /** a nix path or derivation to be used as mkDerivation.src */
+  src,
+  /** is src a .dmg .
+    * These contain multiple folders after extraction
+    * which requires special treatment in the unpackPhase
+    */
+  isDmg ? false,
```

Comment:
Is there no way to detect this? Maybe always set sourceRoot and then afterwards detect the app is just one level deeper if it's not a dmg?

Reply:
That a good suggestion which be an improvement for another PR. I'd like to keep this PR as small a possible without too many possible code paths and thus opportunities for things to go wrong.

Reply:
Also done in [53ce772](https://github.com/NixOS/nixpkgs/commit/53ce772072f61b49b7881fa404c5287a8e2ae2d7)